frontend/taipy-gui/src/components/Taipy/Notification.tsx

```diff
@@ -0,0 +1,31 @@
+import React from "react";
+import Alert from "@mui/material/Alert";  
+import { TaipyBaseProps } from "./utils";
+import { useDynamicProperty } from "../../utils/hooks";
+
+interface NotificationProps extends TaipyBaseProps {
+    severity?: "error" | "warning" | "info" | "success";
+    message: string | (() => string);  // Dynamic string handling
+    variant?: "filled" | "outlined" | "standard";
+    defaultMessage?: string;
+}
+
+const Notification = (props: NotificationProps) => {
+    const { severity = "info", variant = "filled" } = props;
+
+    // Use useDynamicProperty to handle dynamic message and defaultMessage
+    let displayMessage = useDynamicProperty(props.message, props.defaultMessage, undefined);
+
+    // Ensure displayMessage is a string
+    if (typeof displayMessage === "function") {
```

Comment:
I don't understand what your goal is here.
You're not testing if it's a string. You're testing if it's a function...

Reply:
no need for this, it's all taken care of by `useDynamicProperty`